paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentSheetViewModelTest.kt

```diff
@@ -2390,6 +2393,78 @@ internal class PaymentSheetViewModelTest {
         assertThat(isPhoneRequired).isFalse()
     }
 
+    @Test
+    fun `Can complete payment after switching to another LPM from card selection with inline Link signup state`() =
+        runTest {
+            Dispatchers.setMain(testDispatcher)
+
+            val interceptor = spy(FakeIntentConfirmationInterceptor())
+
+            val viewModel = spy(
+                createViewModel(
+                    customer = EMPTY_CUSTOMER_STATE,
+                    intentConfirmationInterceptor = interceptor,
+                    linkState = LinkState(
+                        configuration = LINK_CONFIG,
+                        loginState = LinkState.LoginState.LoggedOut
+                    )
+                )
+            )
+
+            viewModel.primaryButtonUiState.test {
+                skipItems(1)
+
+                viewModel.updateSelection(
+                    PaymentSelection.New.Card(
+                        paymentMethodCreateParams = PaymentMethodCreateParams.createCard(
+                            CardParams(
+                                number = "4242424242424242",
+                                expMonth = 4,
+                                expYear = 2025,
+                                cvc = "501"
+                            )
+                        ),
+                        brand = CardBrand.Visa,
+                        customerRequestedSave = PaymentSelection.CustomerRequestedSave.NoRequest,
+                    )
+                )
+
+                skipItems(1)
+
+                viewModel.onLinkSignUpStateUpdated(
+                    InlineSignupViewState.create(config = LINK_CONFIG).copy(
+                        isExpanded = true,
+                        userInput = UserInput.SignUp(
+                            name = "John Doe",
+                            email = "johndoe@email.com",
+                            phone = "+15555555555",
+                            consentAction = SignUpConsentAction.CheckboxWithPrefilledEmailAndPhone,
+                            country = "US",
+                        )
+                    )
+                )
+
+                skipItems(1)
+
+                viewModel.updateSelection(
+                    PaymentSelection.New.GenericPaymentMethod(
+                        iconResource = R.drawable.stripe_ic_paymentsheet_card_visa,
+                        labelResource = "Bancontact",
+                        paymentMethodCreateParams = PaymentMethodCreateParamsFixtures.BANCONTACT,
+                        customerRequestedSave = PaymentSelection.CustomerRequestedSave.NoRequest,
+                        lightThemeIconUrl = null,
+                        darkThemeIconUrl = null,
+                    )
+                )
+
+                awaitItem()?.onClick?.invoke()
+
+                cancelAndIgnoreRemainingEvents()
```

Comment:
Why do we need this? What events are we not consuming?

Reply:
Thought it was needed since `StateFlow` does not end. Looks like that's not the case. Consumed the remaining event instead where the button is disabled after entering the processing state.